strapi/Dockerfile

```diff
@@ -1,57 +1,66 @@
-# development
-FROM node:16 AS dev
-
-RUN apt-get update && apt-get install -y git \
-    vim \
-    libvips-dev
-
-ARG NODE_ENV=development
-ENV NODE_ENV=${NODE_ENV}
-
-WORKDIR /home/node/app
-COPY package*.json ./
-COPY yarn.lock ./
+FROM node:16-alpine AS build-base
+RUN apk update \
+ && apk add --no-cache build-base \
+                       gcc \
+                       autoconf \
+                       automake \
+                       zlib-dev \
+                       libpng-dev \
+                       vips-dev > /dev/null 2>&1
+WORKDIR /build
+COPY ./package.json ./yarn.lock ./
+# Uncomment if you need any to apply any patches
 COPY patches ./patches
+# copy any installation files for Strapi plugins you want to build. For example:
+# COPY src/plugins/<package-name>/package.json src/plugins/<package-name>/yarn.lock ./src/plugins/<package-name>
+# Or leave empty (do not add anyting) if you don't have any custom plugins (most likely ./plugins directory)
 COPY src/plugins/ceremonies-debtor-list/package.json ./src/plugins/ceremonies-debtor-list/package.json
 COPY src/plugins/ceremonies-debtor-list/yarn.lock ./src/plugins/ceremonies-debtor-list/yarn.lock
 
-RUN yarn install
-COPY . ./
-
-RUN yarn build
-
-EXPOSE 1337
-CMD ["yarn", "develop"]
-
-# production
-FROM node:16 AS prod
+FROM build-base AS install-prod
+RUN yarn config set network-timeout 600000 -g \
+ && yarn install --production
 
-RUN apt-get update && apt-get install -y git \
-    libvips-dev
+FROM install-prod as build
+WORKDIR /build/app
+COPY ./ .
+RUN yarn install --development && yarn build
 
+FROM node:16-alpine AS app-base
+RUN apk add --no-cache vips-dev \
+                       tini \
+ && mkdir -p /home/node/app \
+ && chown node:node /home/node/app
```

Comment:
I would split these into apk add and then mkdir with chown

Reply:
That is interesting notion. Adding one more layer so more layers can be just reused and when packaged to hub keeping it smaller. But on the other hand still adding one more layer, but as it didn't increase change much about the image I split the command into two RUN instructions as it probably makes more sense that way.